backend/hitas/views/apartment_sale.py

```diff
@@ -87,14 +87,23 @@ def create(self, validated_data: dict[str, Any]) -> ApartmentSale:
                     Ownership.objects.select_related("apartment"),
                 ),
                 # Limit the fetched sales to only the first sale,
-                # as we only need that to figure out if the apartment is new
-                prefetch_first_sale("ownerships__apartment__"),
+                # as we only need that to figure out if the apartment is new.
+                # Ignore the sale we just created so that this apartment is treated as new.
+                prefetch_first_sale("ownerships__apartment__", ignore=[instance.id]),
             )
             for owner in owners:
                 cos = create_conditions_of_sale(owners=[owner])
                 if cos:
                     self.context["conditions_of_sale_created"] = True
 
+            apartment.first_purchase_date = instance.purchase_date
+            apartment.save()
+            return instance
+
+        if apartment.latest_purchase_date is None or instance.purchase_date > apartment.latest_purchase_date:
+            apartment.latest_purchase_date = instance.purchase_date
+            apartment.save()
```

Comment:
Is this last commit about something to do with _conditions of sale_ like the name suggests, or is it about setting the purchase dates? 🤔

Reply:
I simply forgot that these fields have to be set (for now), so I put them in the commit. I guess I could've separated them into their own commit.